lib/iris/_representation/cube_summary.py

```diff
@@ -259,8 +259,17 @@ def __init__(self, cube, shorten=False, name_padding=35):
         vector_dim_coords = [
             coord for coord in dim_coords if id(coord) not in scalar_coord_ids
         ]
+        if cube.mesh is None:
```

Comment:
These changes I've just done this for now, as it was irritating me + making debugging hard (not to easily see when there is a mesh or not).

**TODO: pull out these changes (1 commit) and put them in their own PR.**

Reply:
@pp-mo I can see why you done this here, makes total sense.

You've added testing in this PR to cover this change, so I don't really see the advantage in separating concerns here to end up at the same destination.

I think we're inevitably going to have to add extra code (e.g., updating the cube summary) when we integrate UGRID features into `iris`. 

So thanks for making the change, and I'm happy with this as it stands, if you are?

Reply:
Yes if you're happy.
I have already included the minimal test additions + changes I thought necessary.

The need for this really became apparent when I was collapsing the mesh on cubes to allow cube comparisons in the round-trip tests :  The resulting replacement of MeshCoords with AuxCoords is completely invisible -- the cube printout was just the same.

It's worth considering, though, that this is still a "minimal" approach, compared with ways we have considered making a structured cube visible in the past :  It does not show anyt mesh details -- most notably, it **_doesn't_** say what mesh location the cube maps to.  Nor does it let you judge whether 2 cubes _share_ a mesh (e.g. by a mesh name)

Reply:
Yeah, I think we're in a better position to enrich the cube summary. Going back to before Xmas, I was keen to include minimal mesh information in the cube summary... and this is a good (self) use case to push that forwards (seems like)